src/ir_Corona.cpp

```diff
@@ -0,0 +1,569 @@
+    DPRINT(uint64ToString(section));
+    used = matchGeneric(results->rawbuf + offset, results->state + pos,
+                        results->rawlen - offset, kCoronaAcBitsShort,
+                        kCoronaAcHdrMark, kCoronaAcHdrSpace,
+                        kCoronaAcBitMark, kCoronaAcOneSpace,
+                        kCoronaAcBitMark, kCoronaAcZeroSpace,
+                        kCoronaAcBitMark, kCoronaAcSpaceGap, true,
+                        _tolerance + kCoronaTolerance, kMarkExcess, false);
+    if (used == 0) return false;  // We failed to find any data.
+    // short versions section 0 is special
+    if (strict && !IRCoronaAc::validSection(results->state, pos,
+                                            isLong ? section : 3))
+      return false;
+    offset += used;  // Adjust for how much of the message we read.
+    pos += kCoronaAcSectionBytes;  // Adjust by how many bytes of data was read
+    // don't read more data then what we have
+    if (results->rawlen <= offset)
+      break;
+  }
+
+  // Re-check we got the correct size/length due to the way we read the data.
+  if (strict && pos * 8 != kCoronaAcBits && pos * 8 != kCoronaAcBitsShort) {
+    DPRINTLN("strict bit match fail");
+    return false;
+  }
+
+  // Success
+  results->decode_type = decode_type_t::CORONA_AC;
+  results->bits = pos * 8;
+  // No need to record the state as we stored it as we decoded it.
+  // As we use result->state, we don't record value, address, or command as it
+  // is a union data type.
+  return true;
+}
+#endif  // DECODE_CORONA_AC
+
+/// Class constructor for handling detailed Corona A/C messages.
+/// @param[in] pin GPIO to be used when sending.
+/// @param[in] inverted Is the output signal to be inverted?
+/// @param[in] use_modulation Is frequency modulation to be used?
+/// @return An IRCoronaAc object.
+IRCoronaAc::IRCoronaAc(const uint16_t pin, const bool inverted,
+                       const bool use_modulation)
+    : _irsend(pin, inverted, use_modulation) { stateReset(); }
+
+/// Reset the internal state to a fixed known good state.
+/// @note The state is powered off.
+void IRCoronaAc::stateReset(void) {
+  // known good state
+  remote_state[kCoronaAcSectionD0Pos] = kCoronaAcSectionD0Base;
+  remote_state[kCoronaAcSectionD1Pos] = 0x00;  // ensure no unset mem
+  setTemp(kCoronaAcMinTemp);
+  setMode(kCoronaAcModeCool);
+  setFan(kCoronaAcFanAuto);
+  setOnTimer(kCoronaAcTimerOff);
+  setOffTimer(kCoronaAcTimerOff);
+  checksum(remote_state);
+}
+
+/// Get the byte that identifies the section
+/// @param[in] section index of the section 0-2,
+///            3 and above is used as the special case for short message
+/// @return the byte used for the section
+uint8_t IRCoronaAc::getSectionByte(const uint8_t section) {
+  // base byte
+  uint8_t b = kCoronaAcSectionLabelBase;
+  // 2 enabled bits shifted 0-2 bits depending on section
+  if (section >= 3)
+    return 0b10010000 | b;
+  setBits(&b, 4, 4, 0b11 << section);
```

Comment:
`setBits(&b, kUpperNibble, kNibbleSize, 0b11 << section);` makes it more readable.

Reply:
Correct name seems to be; kHighNibble
`setBits(&b, kHighNibble, kNibbleSize, 0b11 << section);`

Reply:
Yeah. Auto complete in the IDE FTW! Sorry I got it wrong (off the top of my head)

Reply:
not at all, just a note to self and future readers.